tools/blockSyncer/upload_download.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+
+	"github.com/go-kit/log"
+	"github.com/thanos-io/objstore"
+	"github.com/thanos-io/objstore/client"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+type Store struct {
+	Bucket       objstore.Bucket
+	TsdbPath     string
+	ObjectKey    string
+	ObjectConfig string
+}
+
+func newstore(tsdbPath, objectConfig, objectKey string) *Store {
+	logger := slog.New(slog.NewJSONHandler(os.Stdout, nil))
```

Comment:
This needs to go to `main`.

Reply:
I think we can declare this line on the `newStore` function and pass it in struct Store{}.